app/src/main/java/com/doublesp/coherence/actions/ListFragmentActionHandler.java

```diff
@@ -70,6 +70,18 @@ public void onFloatingAcitonButtonClick() {
     }
 
     @Override
+    public void onSaveButtonClick() {
+        Plan plan = mIdeaInteractor.getPlan();
+        DatabaseReference keyReference = mListDatabaseReference.push();
+
+        HashMap<String, Object> timestampCreated = new HashMap<>();
+        timestampCreated.put(ConstantsAndUtils.TIMESTAMP, ServerValue.TIMESTAMP);
+        UserList userList = new UserList(plan.getTitle(), plan.getOwner(), timestampCreated);
+        keyReference.setValue(userList);
+        mShoppingListDatabaseReference.child(keyReference.getKey()).setValue(plan);
+    }
```

Comment:
There is one problem here, if user clicks save and then he clicks share or share and save, we are saving this multiple times, we might have to think about handling that case.